### DISCUSSION TITLE
Have the ability to change the async backend

### DISCUSSION THREAD

**Discussion by lewoudar** [flet-dev/flet#990] [Ideas]
Hello everyone,
I saw this [article](https://flet.dev/blog/standalone-flet-web-apps-with-pyodide/#async-support) presenting async support for Flet apps. This is great! Thank you for all your effort.
My proposition is to give the ability to users to select other async backends than asyncio. I'm thinking of [trio](https://trio.readthedocs.io/en/stable/) and [anyio](https://anyio.readthedocs.io/en/stable/) leveraging [structured concurrency](https://en.wikipedia.org/wiki/Structured_concurrency).
The idea will be to run the app with something like the following:

```python
import flet as ft
...
ft.app(main, backend='trio')  # backend should defaults to 'asyncio'
```

**Comment by FeodorFitsner** [comment]
Hm, it could be kind of tricky I guess. I've checked Trio docs and it uses its own synchronization primitives (Lock, channels instead of queues?) as well as classes to work with TCP and files. It means re-implementing portions of Flet internals for each async backend, right?

**Comment by lewoudar** [comment]
Honestly I don't look at Flet code (how yoy handle the async stuff), but it will probably requires a bit of effort, but anyio is pretty straightforward and is compatible with asyncio and trio
